oci/oci_dest.go

```diff
@@ -51,30 +59,55 @@ func (d *ociImageDestination) CanonicalDockerReference() (string, error) {
 	return "", fmt.Errorf("Can not determine canonical Docker reference for an OCI image")
 }
 
+func createManifest(m []byte) ([]byte, string, error) {
+	om := ociManifest{}
+	mt := manifest.GuessMIMEType(m)
+	switch mt {
+	case manifest.DockerV2Schema1MIMEType:
+		return nil, "", fmt.Errorf("not yet implemented")
+	case manifest.DockerV2Schema2MIMEType:
+		if err := json.Unmarshal(m, &om); err != nil {
+			return nil, "", err
+		}
+		om.MediaType = manifest.OCIV1ImageManifestMIMEType
+		b, err := json.Marshal(om)
+		if err != nil {
+			return nil, "", err
+		}
+		return b, om.MediaType, nil
+	case manifest.DockerV2ListMIMEType:
+		return nil, "", fmt.Errorf("not yet implemented")
+	case manifest.OCIV1ImageManifestListMIMEType:
+		return nil, "", fmt.Errorf("not yet implemented")
+	case manifest.OCIV1ImageManifestMIMEType:
+		return m, om.MediaType, nil
+	}
+	return nil, "", fmt.Errorf("Unrecognized manifest media type")
+}
+
 func (d *ociImageDestination) PutManifest(m []byte) error {
 	if err := d.ensureParentDirectoryExists("refs"); err != nil {
 		return err
 	}
-	// TODO(runcom):
-	// make a function to create an OCI manifest from a distribution one which comes here!!!
-	// convert the manifest to OCI v1 - beaware we can also receive a docker v2s1 manifest so we need to
-	// handle this! docker.io/library/busybox it's still on v2s1 for instance
-	digest, err := manifest.Digest(m)
+	ociMan, mt, err := createManifest(m)
```

Comment:
Creating the real manifest so late in `ImageDestination` won’t work if we want to sign manifests; `copy --sign-by` will not have the right manifest to sign.

For now, especially when the conversion is only a matter of changing a single field, I guess this could live in `copy.go`. Or, perhaps, just make this a TODO for now (we can’t sign anyway), I can live with that.

Eventually we will need to make that a library, and I can’t immediately see a clean way. `image.ByConvertingImageToMIMEType` perhaps? That also seems fairly ugly and it is not immediately clear that it can work.  (For that matter, `signature.SignDockerManifest` will need to be replaced/generalized as well.)